crates/ty_python_semantic/resources/mdtest/snapshots/overloads.md_-_Overloads_-_Invalid_-_Overload_without_an_…_-_Regular_modules_(5c8e81664d1c7470).snap

```diff
@@ -17,43 +17,47 @@ mdtest path: crates/ty_python_semantic/resources/mdtest/overloads.md
  3 | @overload
  4 | def func(x: int) -> int: ...
  5 | @overload
- 6 | # error: [invalid-overload] "Overloaded non-stub function `func` must have an implementation"
+ 6 | # error: [invalid-overload] "Overloads for function `func` must be followed by a non-`@overload`-decorated implementation function"
  7 | def func(x: str) -> str: ...
  8 | 
  9 | class Foo:
 10 |     @overload
 11 |     def method(self, x: int) -> int: ...
 12 |     @overload
-13 |     # error: [invalid-overload] "Overloaded non-stub function `method` must have an implementation"
+13 |     # error: [invalid-overload] "Overloads for function `method` must be followed by a non-`@overload`-decorated implementation function"
 14 |     def method(self, x: str) -> str: ...
 ```
 
 # Diagnostics
 
 ```
-error[invalid-overload]: Overloaded non-stub function `func` must have an implementation
+error[invalid-overload]: Overloads for function `func` must be followed by a non-`@overload`-decorated implementation function
  --> src/mdtest_snippet.py:7:5
   |
 5 | @overload
-6 | # error: [invalid-overload] "Overloaded non-stub function `func` must have an implementation"
+6 | # error: [invalid-overload] "Overloads for function `func` must be followed by a non-`@overload`-decorated implementation function"
 7 | def func(x: str) -> str: ...
-  |     ^^^^
+  |     ^^^^ Attempting to call `func` will raise `TypeError` at runtime
```

Comment:
This is also part of the concise message, and I think it it's more confusing than helpful? We could add a "will raise `TypeError` at runtime" message to a lot of lints…

Reply:
> We could add a "will raise `TypeError` at runtime" message to a lot of lints…

heh, and I like to, when it's true! I think it's useful to be clear when something is a theoretical soundness complaint that might or might not cause issues, and when something will actually, definitely, 100% of the time cause a `TypeError` to be raised (assuming our inference is correct, of course). Subclassing a `@final` class is the former category; this is the latter category! There's no way the user could possibly want this behaviour

Reply:
Good point about it being part of the concise message, though. That's too long for a supposedly single-line diagnostic.